src/app/participant/models.py

```diff
@@ -0,0 +1,59 @@
+from app.models import BaseModel
+from google.appengine.ext import ndb
+
+import tinyid
+from app.league.models import LeagueModel
+from app.user.models import UserModel
+
+__author__ = 'Alex'
+
+
+class ParticipantModel(BaseModel):
+    participant_id = ndb.StringProperty(required=True)
+    league_id = ndb.StringProperty(required=True)
+    user_id = ndb.StringProperty(required=True)
+    name = ndb.StringProperty(indexed=False)
+    rating = ndb.FloatProperty()
+
+    @classmethod
+    def generate_id(cls):
+        """ Generate a unique id. """
+        if ndb.in_transaction():
+            tiny_id = tinyid.TinyIDGenerator(namespace='P').generate_tinyid(run_in_transaction=False).upper()
+        else:
+            tiny_id = tinyid.TinyIDGenerator(namespace='P').generate_tinyid(run_in_transaction=True).upper()
+        return '%s-%s' % ('P', tiny_id)
+
+    @classmethod
+    def build_key(cls, participant_id):
+        """ Builds a key in the default namespace. """
+        key = ndb.model.Key(cls.__name__, participant_id)
+        return key
+
+    @classmethod
+    def _post_delete_hook(cls, key, future):
+        participant = key.get()
+        user_key = UserModel.build_key(cls.user_id)
+        league_key = LeagueModel.build_key(participant.league_id, user_key)
+        league = league_key.get()
+        league.update_participant_count(-1)
+
+    def _post_put_hook(self):
+        user_key = UserModel.build_key(self.user_id)
+        league_key = LeagueModel.build_key(self.league_id, user_key)
+        league = league_key.get()
+        league.update_participant_count(1)
```

Comment:
post_put runs *every time* .put() is called on this entity. That means every time you save this participant, you'll be adding +1 to the league's count. i think you could just do this in the create method for this function and it'll be a lot easier. it's tricky to do post-create hooks right since they don't expose a direct api for that